cpp/src/arrow/flight/transport.cc

```diff
@@ -159,6 +160,198 @@ TransportRegistry* GetDefaultTransportRegistry() {
   return &kRegistry;
 }
 
+//------------------------------------------------------------
+// Error propagation helpers
+
+TransportStatus TransportStatus::FromStatus(const Status& arrow_status) {
+  if (arrow_status.ok()) {
+    return TransportStatus{TransportStatusCode::kOk, ""};
+  }
+
+  TransportStatusCode code = TransportStatusCode::kUnknown;
+  std::string message = arrow_status.message();
+  if (arrow_status.detail()) {
+    message += ". Detail: ";
+    message += arrow_status.detail()->ToString();
+  }
+
+  std::shared_ptr<FlightStatusDetail> flight_status =
+      FlightStatusDetail::UnwrapStatus(arrow_status);
+  if (flight_status) {
+    switch (flight_status->code()) {
+      case FlightStatusCode::Internal:
+        code = TransportStatusCode::kInternal;
+        break;
```

Comment:
We have `Status`, `FlightStatusDetail`, `TransportStatus` and according code definitions. Looks to me `FlightStatusDetail` and `TransportStatus` are similar. Why we need them both?

Reply:
I think this PR is okay, not necessary to rework.